core/web-assets/src/main/assets/js/apps/onms-daemons/index.js

```diff
@@ -0,0 +1,92 @@
+require('angular-bootstrap-confirm');
+require('angular-bootstrap-toggle/dist/angular-bootstrap-toggle');
+require('angular-bootstrap-toggle/dist/angular-bootstrap-toggle.css');
+require('angular-ui-router');
+
+const indexTemplate = require('./index.html');
+
+(function () {
+    'use strict';
+
+    var MODULE_NAME = 'onms.daemons';
+
+    angular.module(MODULE_NAME, [
+        'ui.router',
+        'ui.bootstrap',
+        'ui.toggle',
+        'mwl.confirm',
+        'onms.http'
+    ])
+        .config(['$locationProvider', function ($locationProvider) {
+            $locationProvider.hashPrefix('!');
+            $locationProvider.html5Mode(false);
+        }])
+        .config(['$stateProvider', function ($stateProvider) {
+            $stateProvider
+                .state('daemons', {
+                    url: '',
+                    controller: 'ListController',
+                    templateUrl: indexTemplate
+                });
+        }])
+        .controller('ListController', ['$scope', '$http', '$timeout', function ($scope, $http, $timeout) {
+
+            // This is used for the curl command-string in the view
+            $scope.serverUrl = location.protocol + '//' + location.hostname+(location.port ? ':'+location.port: '');
+
+            $scope.daemons = [];
+
+            $scope.lookupDaemonState = function (daemon) {
+                if (daemon.reloadCount > 5) {
+                    daemon.isReloading = false;
+                    daemon.reloadState = "Undefined";
+                    return;
+                }
+                daemon.reloadCount = daemon.reloadCount + 1;
+
+                $http.get('rest/daemons/reload/' + daemon.name + '/').then(function (response) {
+                    if (response.data.reloadState === "Success") {
+                        daemon.isReloading = false;
+                        daemon.reloadState = response.data.reloadState;
+                    } else if (response.data.reloadState === "Failed") {
+                        daemon.isReloading = false;
+                        daemon.reloadState = response.data.reloadState;
+                    } else {
+                        $timeout($scope.lookupDaemonState, 1000, true, daemon);
+                    }
+                });
+            };
+
+            $scope.reloadDaemon = function (daemon) {
+                $http.post('rest/daemons/reload/' + daemon.name + '/')
+                    .success(function (data, status) {
+                        if (status === 204) {
+                            var now = new Date();
+                            daemon.reloadTime = now.getTime() + (now.getTimezoneOffset() * 60000);
+                            daemon.isReloading = true;
+                            daemon.reloadCount = 0;
+                            daemon.reloadState = "Reloading";
+                            $timeout($scope.lookupDaemonState, 1000, true, daemon);
+                        }
+                    });
+            };
```

Comment:
Looking at the reload logic here, there is the following issue.
If the rest endpoint cannot be reached or failed, reloading is not failing immediately, instead the whole ui is hanging in the reloading state. 
In my opinion the reload logic should be implemented in a more robust way:
 - click on reload sets the ui in state reloading
 - if reloading due to post issue failed -> State Failure -> no event was sent
 - if reloading is timing out a "Timeout" failure should be posted
 - if reload succeeds or fails within the 5 seconds the result should be accordingly